sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -1012,6 +1012,196 @@ abstract class CSVSuite
     }
   }
 
+  test("SPARK-37326: Use different pattern to write and infer TIMESTAMP_NTZ values") {
+    withTempDir { dir =>
+      val path = s"${dir.getCanonicalPath}/csv"
```

Comment:
Just write directly to `dir`

Reply:
Oh, I could not because the directory already exists by the time the method is called. `withTempDir` creates a directory, but it needs to be non-existent for Spark to write into.

Reply:
There is `withTempPath` which doesn't create any dirs:
https://github.com/apache/spark/blob/0ba1d3852b3e6f89128403e988d6ca8772d1706c/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/plans/SQLHelper.scala#L62-L66

Reply:
+1 to use `withTempPath`

Reply:
It seems there is a mixture of `withTempDir` and `withTempPath` in this file arguably being used with the same purpose. I will open a PR to fix the tests I added but I am not going to update other occurrences.